### PR TITLE
fix: remove unnecessary horizontal scroll when table columns fit within table widget width and prevent unnecessary re-renders

### LINKED ISSUE #10412: [Bug]: Table styling bugs
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Listing all the bugs here
1. Whenever we have background color for a cell, it applies to the whole table on search.
2. Even if the table data is fits the width, we observe a scroll horizontally. The last column doesn't resize to fit the width. Occurs even if there's less data than the width of column.

[LOOM DEMO](https://www.loom.com/share/6527f28f1cce4c33a08bf56f62ec7955)

### Steps To Reproduce

Background color issue
1. Add a background color to the first column.
2. Search for any of the values in the row.
3. Observe that the color of the remaining rows changes to the color of the first cell.
4. Happens only for 1 search result.

Scroll issue
1. Add a table with data less than width of the table widget.
2. Scroll horizontally in the view mode. 
3. Observe the scroll happening even though size of data is less than the width of the widget

### Environment

Production

### Version

Cloud

### LINKED ISSUE #11671: [Bug]: Background colour is not working properly on Table widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Screenshot 2022-03-07 at 1 37 10 PM

### Steps To Reproduce

1. Drag and drop a table widget
2. Change the background colour 
3. Observe that the colour is not displayed properly

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description

The table widget has a small scroll issue that always occurs even if the columns are perfectly fit and shouldn't scroll
Using HTML default ellipses to fix the background color issue

Fixes #10412
Fixes #11671

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Visual

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/table-widget-scroll-issue 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 56.18 **(0)** | 37.66 **(-0.01)** | 35.7 **(0)** | 56.43 **(0)**
 :red_circle: | app/client/src/widgets/TableWidget/component/AutoToolTipComponent.tsx | 57.14 **(-0.44)** | 30.56 **(-8.15)** | 28.57 **(-4.76)** | 58.06 **(-0.56)**
 :green_circle: | app/client/src/widgets/TableWidget/component/TableStyledWrappers.tsx | 87.64 **(0.14)** | 56.2 **(0)** | 80.7 **(0)** | 87.64 **(0.14)**</details>

### CODE REVIEW

**Review by sbalaji1192 — APPROVED** [review]
@Tooluloope Could you improve the PR title?
something like - remove unnecessary horizontal scroll when table columns fit within table widget width.

**Comment by sbalaji1192** [comment]
For some reason, UI result check is not updated even though ok to test was successful. retriggering

**Comment by Tooluloope** [comment]
@sbalaji1192, I realized some rerenders, so I felt I should optimize it, 
Could you help review it?

File: app/client/src/widgets/TableWidget/component/AutoToolTipComponent.tsx [hunk 1]
```diff
@@ -115,10 +114,20 @@ function AutoToolTipComponent(props: Props) {
         ) : (
           props.children
         )}
+        {useToolTip && props.children && "..."}
```
[the remark this answers is not in the document]
Comment (Tooluloope) [reply app/client/src/widgets/TableWidget/component/AutoToolTipComponent.tsx:117]: It's still in QA pipeline

File: app/client/src/widgets/TableWidget/component/AutoToolTipComponent.tsx [hunk 2]
```diff
@@ -82,10 +83,8 @@ function AutoToolTipComponent(props: Props) {
     const element = ref.current;
     if (element && element.offsetWidth < element.scrollWidth) {
       updateToolTip(true);
```
[the remark this answers is not in the document]
Comment (Tooluloope) [reply app/client/src/widgets/TableWidget/component/AutoToolTipComponent.tsx]: @sbalaji1192 I changed to the default HTML ellipsis, Not sure why this wasn't used from scratch

**Review by sbalaji1192 — APPROVED** [review]

**Comment by YogeshJayaseelan** [comment]
@Tooluloope - Tested this PR and please find the status below

Issue 1 - Still open 

The background colour applied to column 1 appears on all of the remaining rows on the search 

https://www.loom.com/share/7a7f453dffbd4b87acac00cdf49ca2f9

Issue 2 - Now the horizontal scroll is not happening on the view mode. But, however, observed a wired scenario in the view mode - When the user tries to scroll by keeping the mouse focus on any of the tables the entire page gets scrolled slightly 

https://www.loom.com/share/73aeb329ed5a4d428775f96f87242a92

**Comment by Tooluloope** [comment]
@YogeshJayaseelan The background color and search issue are not being handled in this PR, I'm handling the second part which is the scrolling part.

Secondly, the horizontal scroll you're experiencing is not related to this PR has this is seen on Prod and release

**Comment by YogeshJayaseelan** [comment]
@Tooluloope - Thanks for clarifying 

Issue 1 - Background colour issue will be handled in separate PR

Issue 2 - **Horizontal scroll on the table widget view mode is fixed** and we will raise a separate issue for the page scroll

**Comment by Tooluloope** [comment]
@dilippitchika Should I just Deal with the background color issue?

**Comment by dilippitchika** [comment]
@Tooluloope background color, yes let's deal it in a separate issue. I belive there is one for it

**Comment by Tooluloope** [comment]
@dilippitchika okay thanks

**Pull request merged**